crates/rbuilder/src/live_builder/block_output/relay_submit.rs

```diff
@@ -415,16 +402,29 @@ pub async fn run_submit_to_relays_job_and_metrics(
     }
 }
 
+fn log_validation_error(
+    err: ValidationError,
+    level: Level,
+    validation_use: &str,
+    redactor: Arc<dyn ValidationErrorRedactor>,
+) {
+    dynamic_event!(level,err = ?redactor.redact(err), validation_use,"Validation failed");
+}
+
+/// Validates the blocks handling any logging.
+/// Answers if the block was validated ok.
 async fn validate_block(
     slot_data: &MevBoostSlotData,
     signed_submit_request: &SubmitBlockRequest,
     block: SealedBlock,
     config: &SubmissionConfig,
     cancellation_token: CancellationToken,
-) -> Result<(), ValdationError> {
+    validation_use: &str,
+    redactor: Arc<dyn ValidationErrorRedactor>,
```

Comment:
It makes me sad to see every fns params spammed with `redactor`, this feels like it will leak to absolutely everything. 

Will there ever be the need to redact some parts of the app and not others? I know you like to avoid them (I do too), but this may be a good use case for a lazy_static global, for the same reason logging libraries don't require passing the logging config to everywhere you want to log something.

Reply:
I don't like singletons/global stuff (they hide dependencies) but you might be right that for these I could relax my OCDness.